postgres_base_block.py

```diff
@@ -1,14 +1,78 @@
 from nio.block.base import Block
-from nio.util.discovery import discoverable
-from nio.properties import VersionProperty
+from nio.command import command
+from nio.util.discovery import not_discoverable
+from nio.properties import (VersionProperty, StringProperty, IntProperty,
+                            BoolProperty, PropertyHolder, ObjectProperty)
+from psycopg2 import connect
 
 
-@discoverable
-class Example(Block):
+class AuthCreds(PropertyHolder):
+    username = StringProperty(title="Username", default="", allow_none=False)
```

Comment:
Username should also be allow_none True. Basically, support having no authentication on a database